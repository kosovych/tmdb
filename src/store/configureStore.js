import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'
import axios from '../axios';

import logic from './logics';
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {
    axios
  }

  const logicMiddleware = createLogicMiddleware(logic, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
