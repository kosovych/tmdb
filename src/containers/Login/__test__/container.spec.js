import React from 'react'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'

import diveTo from 'Utils/testHelpers/diveToEnzyme'
import LoginWrapper, { LoginContainer, handleSubmit } from '../container'

jest.mock('Store/auth/actions', () => ({
  loginSubmit: jest.fn()
}))

jest.mock('Store/auth/selectors', () => ({
  authLoadingSelector: jest.fn(() => false),
  errorMessageSelector: jest.fn(() => null),
  sessionIdSelector: jest.fn(() => '')
}))


describe('Login Container', () => {
  const store = configureStore()({})
  store.dispatch = jest.fn()

  const wrapper = shallow(<LoginWrapper store={store} />)
  const container = diveTo(wrapper, LoginContainer)

  it('renders Login component', () => {
    expect(container).toMatchSnapshot()
  })

  describe('handleSubmit()', () => {
    it('calls login() with correct params', () => {
      const login = jest.fn()
      const value = { password: 'password', username: 'username' }
      const props = { login }
      handleSubmit(value, { props })
      expect(login).toHaveBeenCalledWith(value)
    })
  })
})
