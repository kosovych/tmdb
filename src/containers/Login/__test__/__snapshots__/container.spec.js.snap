// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container renders Login component 1`] = `
<Login
  dirty={false}
  errorMessage={null}
  errors={Object {}}
  getFieldHelpers={[Function]}
  getFieldMeta={[Function]}
  getFieldProps={[Function]}
  handleBlur={[Function]}
  handleChange={[Function]}
  handleReset={[Function]}
  handleSubmit={[Function]}
  initialErrors={Object {}}
  initialTouched={Object {}}
  initialValues={
    Object {
      "password": "",
      "username": "",
    }
  }
  isSubmitting={false}
  isValid={true}
  isValidating={false}
  loading={false}
  login={[Function]}
  registerField={[Function]}
  resetForm={[Function]}
  sessionId=""
  setErrors={[Function]}
  setFieldError={[Function]}
  setFieldTouched={[Function]}
  setFieldValue={[Function]}
  setFormikState={[Function]}
  setStatus={[Function]}
  setSubmitting={[Function]}
  setTouched={[Function]}
  setValues={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  submitCount={0}
  submitForm={[Function]}
  touched={Object {}}
  unregisterField={[Function]}
  validateField={[Function]}
  validateForm={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validateOnMount={false}
  values={
    Object {
      "password": "",
      "username": "",
    }
  }
/>
`;
